src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -420,27 +422,67 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
             Contract.Assert(nestedProperty != null);
             Contract.Assert(resource != null);
             Contract.Assert(readContext != null);
-
+            object value = null;
             if (readContext.IsDeltaOfT)
```

Comment:
This conditional originally existed in order to raise an exception for nested resourceSet nav prop in an update, since that previously wasn't supported. Now that it's supported, I believe we can replace lines 425-440 with:
   value = ReadNestedResourceInline(resourceWrapper, nestedProperty.Type, readContext);
 And remove the new ReadDeltaNestedResourceInline (which seems to have already had logic for reading deltas)